cmd/snap/cmd_run.go

```diff
@@ -99,22 +126,43 @@ func snapRun(snapApp, command string, args []string) error {
 		return fmt.Errorf("cannot find app %q in %q", appName, snapName)
 	}
 
-	// build command to run
+	return runSnapConfine(info, app.SecurityTag(), snapApp, command, args)
+}
+
+func snapRunHook(snapName, hookName, revision string) error {
+	info, err := getSnapInfo(snapName, revision)
+	if err != nil {
+		return err
+	}
+
+	hook := info.Hooks[hookName]
+	if hook == nil {
+		return fmt.Errorf("cannot find hook %q in %q", hookName, snapName)
+	}
+
+	hookBinary := filepath.Join(info.HooksDir(), hook.Name)
+
+	return runSnapConfine(info, hook.SecurityTag(), hookBinary, "", nil)
```

Comment:
About the hookBinary. I think what we want is some stronger validation inside `snap-confine` what it actually launcher. The security tag is well structured so `snap-confine` will do something like `_, snap, app := strings.SplitN(securityTag, ".", 3)` and check if the hookBinary is inside /snap/$snap/*/** - that will prevent the issue that anything can be run under a snap security confinement manually.